pkg/containerruntime/containerd.go

```diff
@@ -172,3 +170,132 @@ systemctl enable --now containerd
 func (eng *Containerd) String() string {
 	return containerdName
 }
+
+type containerdConfigManifest struct {
+	Version int                    `toml:"version"`
+	Metrics *containerdMetrics     `toml:"metrics"`
+	Plugins map[string]interface{} `toml:"plugins"`
+}
+
+type containerdMetrics struct {
+	Address string `toml:"address"`
+}
+
+type containerdCRIPlugin struct {
+	Containerd   *containerdCRISettings `toml:"containerd"`
+	Registry     *containerdCRIRegistry `toml:"registry"`
+	SandboxImage string                 `toml:"sandbox_image,omitempty"`
+}
+
+type containerdCRISettings struct {
+	Runtimes map[string]containerdCRIRuntime `toml:"runtimes"`
+}
+
+type containerdCRIRuntime struct {
+	RuntimeType string      `toml:"runtime_type"`
+	Options     interface{} `toml:"options"`
+}
+
+type containerdCRIRuncOptions struct {
+	SystemdCgroup bool
+}
+
+type containerdCRIRegistry struct {
+	Mirrors map[string]containerdRegistryMirror `toml:"mirrors"`
+	Configs map[string]containerdRegistryConfig `toml:"configs"`
+}
+
+type containerdRegistryMirror struct {
+	Endpoint []string `toml:"endpoint"`
+}
+
+type containerdRegistryConfig struct {
+	TLS  *containerdRegistryTLSConfig `toml:"tls"`
+	Auth *AuthConfig                  `toml:"auth"`
+}
+
+type containerdRegistryTLSConfig struct {
+	InsecureSkipVerify bool `toml:"insecure_skip_verify"`
+}
+
+// AuthConfig is a COPY of github.com/containerd/containerd/pkg/cri/config.AuthConfig.
```

Comment:
https://github.com/containerd/containerd/pkg/cri/config.AuthConfig returns 404 `Not Found`

Maybe we can just replace it with https://github.com/containerd/containerd/blob/v1.5.9/pkg/cri/config/config.go#L126

Reply:
because `github.com/containerd/containerd/pkg/cri/config.AuthConfig` is Go type identity, not an URL like https://github.com/containerd/containerd/blob/v1.5.9/pkg/cri/config/config.go#L126

Reply:
Yes, I'm completely aware of that but IMO having an accessible URL here can be better since it's documentation. But fine either way :)